src/MessageTrait.php

```diff
@@ -177,7 +204,20 @@ private function setHeaders(array $headers)
     private function trimHeaderValues(array $values)
     {
         return array_map(function ($value) {
+            if (!is_string($value)) {
+                throw new \InvalidArgumentException('Header value must be a string.');
+            }
             return trim($value, " \t");
         }, $values);
     }
+
+    /**
+     * @param string $header
```

Comment:
those param annotations are not correct. the whole point of those methods is that they can accept anything and validate it. so you can remove phpdocs like this.